src/main/java/de/tum/in/www1/artemis/domain/quiz/QuizSubmission.java

```diff
@@ -29,8 +29,9 @@ public class QuizSubmission extends Submission {
     @JsonView(QuizView.After.class)
     private Double scoreInPoints;
 
-    @Column(name = "quiz_batch")
-    private Long quizBatch;
+    @ManyToOne
+    @JoinColumn(name = "quiz_batch", referencedColumnName = "id")
+    private QuizBatch quizBatch;
```

Comment:
It was an explicit decision to use `Long quizBatch` here to avoid issues with additional Hibernate database fetched, because ManyToOne cannot be implemented with @Lazy and we typically don't need the quizBatch when loading the quizSubmission.

Unfortunately @bottbenj did not document this design decision here and your changes would lead to significant performance issues.

Please revert this change and try to implement the bugfix without this change

Reply:
I've removed this relation as it is not necessary. I've removed the relation on the QuizBatch class as well.